src/java.base/share/classes/java/lang/Module.java

```diff
@@ -268,13 +269,35 @@ static void implAddEnableNativeAccessAllUnnamed() {
         ALL_UNNAMED_MODULE.enableNativeAccess = true;
     }
 
-    /**
-     * Returns true if module m can access restricted methods.
+    /***
+     * Enables <a href="foreign/package-summary.html#restricted">native access</a>
+     * for this module if the caller module already has native access.
+     * <p>
+     * This method is <a href="foreign/package-summary.html#restricted"><em>restricted</em></a>.
+     * Restricted methods are unsafe, and, if used incorrectly, their use might crash
+     * the JVM or, worse, silently result in memory corruption. Thus, clients should refrain from depending on
+     * restricted methods, and use safe and supported functionalities, where possible.
+     *
+     * @throws IllegalCallerException If the caller module doesn't have native access.
+     */
+    @PreviewFeature(feature=PreviewFeature.Feature.FOREIGN)
+    @CallerSensitive
+    public void enableNativeAccess() {
```

Comment:
Most modification methods in `Module` requires caller to be from this module, or the package open to the caller in case of adding opens. This method's access control doesn't appear in line with those designs.

Reply:
In this case, it requires the caller to already have native access - similar to how you can open a package from an other module to a third module if the package is already open to your module, you can enable native access for a third module if you already have native access.

Reply:
I understand, but I would anticipate a module to only enable native access of one that it reads, or one that it has a module layer controller on. Enabling native access for any module out of the blue sounds suspicious.

Reply:
It is only granting the permission - it does not force the module to actually use it.

To use an analogy: While I can give you the permission to enter my home, that doesn't mean you would have the obligation to use it or visit me. In fact, there is not much you could do against me giving you my permission. If you don't like it, you would simply not use it.

The same relationship applies here - one module granting a permission (that the module holds itself), and the other COULD make use of it - but is in no way obligated to do so.

----

And the same thing applies to specifying `--enable-native-access` on the command line.  
The administrator grants the named modules the permission to use restricted methods - but that doesn't change how a module works that simply never calls restricted methods.

Reply:
I guess you should add the security label back, so security devs can determine the proper access controls for this method.